src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java

```diff
@@ -32,98 +28,111 @@
+            @Override
+            protected Table createTable(final Schema schema) {
+                return schema.createTableBuilder(getUniqueIdentifier())//
+                        .column("my_column", type).build();
+            }
         };
     }
 
-    @Override
-    protected String createBooleanVirtualSchemaTableWithSingleTrueValue() throws SQLException {
-        final Table table = postgresSchema.createTableBuilder(POSTGRES_TABLE_NAME)//
-                .column("boolean", "boolean").build()//
-                .insert(true);
-        this.createdTables.add(table);
-        refreshVirtualSchema();
-        return virtualSchema.getFullyQualifiedName() + "." + POSTGRES_TABLE_NAME;
+    protected Connection createExasolConnection() throws SQLException {
+        return SETUP.getExasolContainer().createConnection();
     }
 
     @Override
-    protected VirtualSchemaTestTable<String> createStringVirtualSchemaTable() throws SQLException {
+    protected SingleValueVirtualSchemaTestTable<String> createStringVirtualSchemaTable() {
         return createTestTable("VARCHAR(500)");
     }
 
-    private void truncateTable(final Table table) throws SQLException {
-        SETUP.getPostgresqlStatement().executeUpdate("TRUNCATE TABLE " + table.getFullyQualifiedName());
-    }
-
-    @Override
-    protected void deleteCreatedTables() {
-        this.createdTables.forEach(Table::drop);
-        this.createdTables.clear();
+    private static abstract class PostgresVirtualSchemaTestTable implements VirtualSchemaTestTable {
+        private final VirtualSchema virtualSchema;
+        private final Table table;
+        private final Schema postgresqlSchema;
+
+        public PostgresVirtualSchemaTestTable() {
+            this.postgresqlSchema = SETUP.getPostgresFactory().createSchema(getUniqueIdentifier());
+            this.table = createTable(this.postgresqlSchema);
+            this.virtualSchema = SETUP.createVirtualSchema(this.postgresqlSchema.getName(), Map.of());
+        }
+
+        protected abstract Table createTable(Schema schema);
+
+        @Override
+        public String getFullyQualifiedName() {
+            return this.virtualSchema.getFullyQualifiedName() + "." + this.table.getName();
+        }
+
+        @Override
+        public void drop() {
+            this.virtualSchema.drop();
+            this.table.drop();
+            this.postgresqlSchema.drop();
+        }
+
+        public Table getTable() {
+            return this.table;
+        }
     }
 
-    protected Connection createExasolConnection() throws SQLException {
-        return SETUP.getExasolContainer().createConnection();
+    private static abstract class SingleValuePostgresVirtualSchemaTestTable<T> extends PostgresVirtualSchemaTestTable
+            implements SingleValueVirtualSchemaTestTable<T> {
+        @Override
+        public void initializeSingleRow(final T value) throws SQLException {
```

Comment:
this implementation should be move to the super class.

Reply:
no, it's an implementation detail.

Reply:
are other implementations, eg, for MySQL, use a different approach here if they want to have a reset a SingleRowTable? I believe it's gonna be quite the same, `truncate(); insertSingleRow();`
or you think this might change?

Reply:
ok